gql/comparator_test.go

```diff
@@ -0,0 +1,279 @@
+package gql
+
+import "testing"
+
+func TestOperators(t *testing.T) {
+	tests := []struct {
+		description          string
+		operation            NewListOperation
+		arguments            map[string]interface{}
+		operand              interface{}
+		wantInvalidOperation bool
+		want                 bool
+	}{
+		{
+			description: "string equal valid and true",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     "a",
+			want:        true,
+		},
+		{
+			description: "string equal valid and false",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     "b",
+			want:        false,
+		},
+		{
+			description: "string equal invalid, mismatched operand",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": "a"},
+			operand:     3,
+			want:        false,
+		},
+		{
+			description:          "NewEqualComparator invalid argument, nil",
+			operation:            NewEqualComparator,
+			arguments:            nil,
+			wantInvalidOperation: true,
+		},
+		{
+			description:          "NewEqualComparator invalid argument, missing value",
+			operation:            NewEqualComparator,
+			arguments:            map[string]interface{}{"nothing": "a"},
+			wantInvalidOperation: true,
+		},
+		{
+			description:          "NewEqualComparator invalid argument, float",
+			operation:            NewEqualComparator,
+			arguments:            map[string]interface{}{"Value": 3.14},
+			wantInvalidOperation: true,
+		},
+		{
+			description: "int equal valid and true",
+			operation:   NewEqualComparator,
+			arguments:   map[string]interface{}{"Value": 1},
+			operand:     1,
+			want:        true,
+		},
+		{
+			description: "int equal valid and false",
```

Comment:
there's no reason for these to not work , but to be thorough, can you add test cases for 0 & a negative number too?

Reply:
sure